uc_reports/uc_reports.admin.inc

```diff
@@ -484,8 +497,30 @@ function uc_reports_products_custom() {
     $query->addField('c', 'totalcount');
   }
 
+  if (module_exists('uc_stock')) {
+    $query->leftJoin('uc_product_stock', 's', 'n.nid = s.nid');
+    switch ($args['stock_tracking']) {
+      case 'tracked':
+        $query->condition('s.active', 1);
+        break;
+
+      case 'untracked':
+        $query->condition('s.active', 1, '!=');
+        break;
+
+      default:
+        // No condition for 'all' or no selection.
+    }
+  }
+
   $products = $query->execute();
   foreach ($products as $product) {
+    if ($args['sold_only']) {
+      //Throw away rows that don't qualify.
+      if (empty($product['sold'])) {
```

Comment:
This could be made more compact by using && in the first if statement
```php
if ($args['sold_only'] && empty($product['sold'])) {
```

Reply:
Yup.